streaming/src/test/scala/org/apache/spark/streaming/ReceivedBlockHandlerSuite.scala

```diff
@@ -174,6 +186,120 @@ class ReceivedBlockHandlerSuite
     }
   }
 
+  test("BlockManagerBasedBlockHandler - count messages") {
+    // ByteBufferBlock-MEMORY_ONLY
+    testRecordcount(true, StorageLevel.MEMORY_ONLY,
```

Comment:
This is so much better! But I realize that there is still duplication. These two tests are line by line exact. So I think these set of lines can be put in a function which just takes one parameter, isBlockManagedBasedBlockHandler. You can keep the name as `testRecordCounts`.